### DISCUSSION TITLE
Schedule application at N Hz

### DISCUSSION THREAD

**Discussion by ugo94490** [nasa/cFS#277] [Q&A] [answered]
Hello everyone, I am currently learning how to use the scheduler_lab and I have some questions.

## Context

I created an application that sends me a simulated temperature between 30 and 40 °C.

I would like this application to be triggered automatically at a regular interval.

I saw that the scheduler_lab application allowed me to do what I wanted.

So I added the corresponding message_id to my application to trigger the temperature sending using the scheduler.

So far everything is working fine, my application is sending the temperature at a regular interval of 1 second.

![Capture](https://user-images.githubusercontent.com/51440330/123091941-74931800-d419-11eb-9df9-49b7dca4eb29.PNG)

I noticed that the scheduler was itself clocked compared to the cf_time application which sends "Wake-Up" packets only every second.

## Question

I would like to know if it was possible to change the interval of sending the "Wake-Up" packet so that my scheduler can send a packet every 30 Hz for example ?

I guess I just need to change a #define somewhere but after a long search I can't find the source of this "timer"

**Comment by skliper** [comment, marked as answer]
If you are interested in using a more full-featured scheduler that supports higher than 1Hz rates it can be found at https://github.com/nasa/sch.  The "sch_lab" application is really just a trivial example, and as currently designed/implemented really is limited to 1 Hz.  To make it run faster than 1 Hz you could adjust the 1Hz (aka 1PPS) time sync "pulse", but that would have other impacts since the system is designed around it's current rate/behavior.  Alternatively you could modify sch_lab to have an additional higher rate timer, but that's basically what the sch app referenced above does.

**Reply by ugo94490** [reply]
Thank you very much, I looked into it quickly but I get some errors when compiling. 

I don't have time to look into it more at the moment but I'll post an update early next week to say if I've solved these errors and how.

**Reply by ugo94490** [reply]
Here are the screenshots of the errors.

Is it normal and it's up to me to fix these errors or since the SCH directory has not been updated for a long time it is no longer compatible with cFS in its current state?

![Error_1](https://user-images.githubusercontent.com/51440330/123246838-c26d5600-d4d5-11eb-96cd-0b2b5cdda0c2.PNG)
![Error_2](https://user-images.githubusercontent.com/51440330/123246848-c600dd00-d4d5-11eb-9380-79c16340436a.PNG)
![Error_3](https://user-images.githubusercontent.com/51440330/123246859-c8fbcd80-d4d5-11eb-8740-27641b718410.PNG)

**Reply by skliper** [reply]
The open source app releases typically lag the core so the trade is to either get a "pre-release" or internal version of the app that works with the current core via a software usage agreement of some sort (whatever agreement/coordination/contacts/mechanisms you may have in place with NASA), use a version of the core that's compatible with the latest open source app release, update the app yourself, or wait for the open source release of the app.

**Comment by skliper** [comment]
Heads up - https://github.com/nasa/sch_lab/pull/104 updates sch_lab to be able to run at faster than 1Hz.
